scripts/lint.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+
+# scripts/lint.sh -*-shell-*-
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Lint all files in the project.
+# Expected to be run from the root of the project. 
+# Expected tools to be installed on the system. A pre-check is done to ensure all tools are installed.
+
+set -e
+set -x
```

Comment:
Remove the execution trace option? 
It's very noisy to use the script with this set.